controller/engine_controller.go

```diff
@@ -973,11 +977,24 @@ func (m *EngineMonitor) refresh(engine *longhorn.Engine) error {
 			m.logger.Debug("Cannot restore the backup for engine since it is still in the backoff window")
 			return nil
 		}
+
+		if isReachedLimit, err := m.isReachedConcurrentVolumeBackupRestoreLimit(); err != nil {
+			m.logger.WithError(err).Warn("Failed to check concurrent volume backup restore limit")
+			return nil
+
+		} else if isReachedLimit {
+			m.logger.Debugf("Reached the concurrent volume backup restore limit of %v, retry later", types.SettingNameConcurrentBackupRestorePerNodeLimit)
+			m.restoreBackoff.Next(engine.Name, time.Now())
+			return nil
+		}
```

Comment:
You can add this part to `aquireRestoringCounter`, but bypass this check if the volume is DR.

Reply:
We need to return this, so keep this as it is.

Reply:
You are right!